### DISCUSSION TITLE
Compute genets: What does it do?

### DISCUSSION THREAD

**Discussion by andieich** [cnr-isti-vclab/TagLab#138] [General]
I am using Taglab to track individual colonies in time. I'm using a mix of the automatic compute manual and automatic matches, which works quite well. I saw the function `compute genets` and did not find documentation. What does it do?

**Comment by maxcorsini** [comment]
Compute genets tracks the matching through the time. More specifically, an unique id is given to a match, and then this is used to track it in the subsequent time points. Let's try to compute it and then save it as a SVG. This visualize the evolution of each genet through the time.

**Reply by andieich** [reply]
Thanks for the info. Sice I am currently using this feature to track colonies over time, I want to understand it better... Aren't the IDs also calculated automatically when I do the automatic/manual matches? At least, I see the unique IDs in the comparison table and when I export the comparisons.

**Comment by ponchio** [comment]
Mathematically each genet is assigned to the connected component of the graph, where each blob is a node and each match is an edge. (if you make many to many match they are all connected).

The ID is recomputed each time you modify a match but also when you add/remove blobs, the ID is not stable. 

We could not figure out a sensible strategy to keep the ID stable.

**Comment by andieich** [comment]
Thanks for the quick reply. So I can use the IDs once I've exported them for further analysis after automatic/manual matching but if I change something and re-export, the IDs would change, correct?

**Comment by ponchio** [comment]
Yes, it would change.
